snapcraft/plugins/rust.py

```diff
@@ -152,6 +152,13 @@ def _fetch_rust(self):
                   '--disable-sudo', '--save'] + options)
 
     def _fetch_deps(self):
+        
+        source_subdir = getattr(self.options, 'source_subdir', None)
```

Comment:
@roxyd Sergio made a mistake there. On the if block, it should be
`sourcedir = os.path.join(self.sourcedir, self.options.source_subdir)`
I'm not sure if that would solve the error you are getting though.